packages/block-editor/src/components/block-tools/block-contextual-toolbar.js

```diff
@@ -18,43 +18,50 @@ import BlockToolbar from '../block-toolbar';
 import { store as blockEditorStore } from '../../store';
 
 function BlockContextualToolbar( { focusOnMount, isFixed, ...props } ) {
-	const { blockType, hasParents, showParentSelector } = useSelect(
-		( select ) => {
-			const {
-				getBlockName,
-				getBlockParents,
-				getSelectedBlockClientIds,
-			} = select( blockEditorStore );
-			const { getBlockType } = select( blocksStore );
-			const selectedBlockClientIds = getSelectedBlockClientIds();
-			const selectedBlockClientId = selectedBlockClientIds[ 0 ];
-			const parents = getBlockParents( selectedBlockClientId );
-			const firstParentClientId = parents[ parents.length - 1 ];
-			const parentBlockName = getBlockName( firstParentClientId );
-			const parentBlockType = getBlockType( parentBlockName );
+	const {
+		blockType,
+		hasParents,
+		isBlockSelected,
+		showParentSelector,
+	} = useSelect( ( select ) => {
+		const {
+			getBlockName,
+			getBlockParents,
+			getSelectedBlockClientIds,
+		} = select( blockEditorStore );
+		const { getBlockType } = select( blocksStore );
+		const selectedBlockClientIds = getSelectedBlockClientIds();
+		const selectedBlockClientId = selectedBlockClientIds[ 0 ];
+		const parents = getBlockParents( selectedBlockClientId );
+		const firstParentClientId = parents[ parents.length - 1 ];
+		const parentBlockName = getBlockName( firstParentClientId );
+		const parentBlockType = getBlockType( parentBlockName );
 
-			return {
-				blockType:
-					selectedBlockClientId &&
-					getBlockType( getBlockName( selectedBlockClientId ) ),
-				hasParents: parents.length,
-				showParentSelector:
-					hasBlockSupport(
-						parentBlockType,
-						'__experimentalParentSelector',
-						true
-					) && selectedBlockClientIds.length <= 1,
-			};
-		},
-		[]
-	);
+		return {
+			isBlockSelected: !! selectedBlockClientIds.length,
+			blockType:
+				selectedBlockClientId &&
+				getBlockType( getBlockName( selectedBlockClientId ) ),
+			hasParents: parents.length,
+			showParentSelector:
+				hasBlockSupport(
+					parentBlockType,
+					'__experimentalParentSelector',
+					true
+				) && selectedBlockClientIds.length <= 1,
+		};
+	}, [] );
 
 	if ( blockType ) {
 		if ( ! hasBlockSupport( blockType, '__experimentalToolbar', true ) ) {
 			return null;
 		}
 	}
 
+	if ( ! isBlockSelected ) {
```

Comment:
This will cause a vertical height shift in the other editors when selecting the title. Maybe we should ensure consistent padding in general when the top toolbar is expected vs just for the navigation editor?

https://user-images.githubusercontent.com/1270189/134582288-099cd79d-149f-46d8-a354-c1e359fa1716.mp4

Reply:
> This will cause a vertical height shift in the other editors when selecting the title. Maybe we should ensure consistent padding in general when the top toolbar is expected vs just for the navigation editor?

I agree, it would be better to keep it consistent. With that in mind, I'm inclined to think that we should keep the empty placeholder in the mobile view.

1. Even with an animation, simply removing it is disruptive to the editing experience, particularly in that some times — depending on where you click the Block — you would have to track back to the Block you clicked after selecting it.
![Kapture 2021-09-27 at 13 39 24](https://user-images.githubusercontent.com/1157901/134950350-d6a34d26-f2a9-4c4c-81ed-f66b492ae7ed.gif)
2. If we were prevent the toolbar from rendering, we would have to add a margin in its place in order to keep content position consistent; since the toolbar sits outside of the editor content, this margin would expose a background that is outside the editor.
<img width="418" alt="Screen Shot 2021-09-27 at 13 44 57" src="https://user-images.githubusercontent.com/1157901/134951320-98bf4bc9-44b9-4c66-9877-41828bdba9d3.png">
In order to cover that up, we would need access to theme colors, as we can't assume that the editor will always have a white background. eg. we'd have to use `#d1e4dd` to cover it if the user is using TT1 — this extrapolates to other themes that implement custom background colors.

I'd like to hear @javierarce's thoughts on this, as I might be missing why the removal is necessary in mobile in the first place.

Reply:
Couldn't the bottom toolbar have a fixed position? That way, the content below won't move when it appears (from my point of view, it's ok not to maintain a constant bottom margin in this case).

This is precisely how this branch works when the user scrolls down and clicks on a block (this happens from second 8 in the following video)

https://user-images.githubusercontent.com/4933/135081084-6c775a50-5739-467a-b9a6-e01e76d71395.mp4